javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/AbstractJavaParserContext.java

```diff
@@ -143,27 +144,29 @@ public SymbolReference<? extends ResolvedValueDeclaration> solveSymbolInParentCo
 
         // First check if there are any pattern expressions available to this node.
         Context parentContext = optionalParentContext.get();
-        if(parentContext instanceof BinaryExprContext || parentContext instanceof IfStatementContext) {
```

Comment:
The changes in this method are all to keep type solving logic for patterns contained. See a comment about this below in `SwitchEntryContext`